src/lightning_app/utilities/app_helpers.py

```diff
@@ -402,3 +401,35 @@ def default(self, obj: Any) -> Any:
         if callable(getattr(obj, "__json__", None)):
             return obj.__json__()
         return json.JSONEncoder.default(self, obj)
+
+
+class Logger:
+
+    """This class is used to improve the debugging experience."""
+
+    def __init__(self, name: str):
+        self.logger = logging.getLogger(name)
+        self.level = None
+
+    def info(self, msg, *args, **kwargs):
+        self.logger.info(msg, *args, **kwargs)
+
+    def warn(self, msg, *args, **kwargs):
+        self._set_level()
+        self.logger.warn(msg, *args, **kwargs)
+
+    def debug(self, msg, *args, **kwargs):
+        self._set_level()
+        self.logger.debug(msg, *args, **kwargs)
+
+    def error(self, msg, *args, **kwargs):
+        self._set_level()
+        self.logger.error(msg, *args, **kwargs)
+
+    def _set_level(self):
```

Comment:
Despite it's name, this does not set the level. Because if level is not None, it will never be set. What's the reasoning here? Why have a level attribute if it won't ever affect the logging level when set?

Reply:
My bad, here is the final version.

```python
    def _set_level(self):
        """Lazily set the level once set by the users."""
        if self.level is None:
            self.level = logging.DEBUG if bool(int(os.getenv("DEBUG", "0"))) else logging.INFO
            self.logger.setLevel(self.level)
```

The goal is to set debug level if the environment is set only when the logger is actually used.